substratevm/src/com.oracle.svm.agent/README.md

```diff
@@ -0,0 +1,77 @@
+# Debugging the agent with GDB
+
+The native image agent is a native library that uses JVMTI.
+In this document you will learn how to debug the agent.
+
+Debugging an agent requires building it with its debug symbols.
+To do that, execute:
+
+```bash
+$ native-image -g --macro:native-image-agent-library
+```
+
+Next, in order to debug the agent,
+the java process that runs agent needs to be executed from within GDB,
+assuming that the `java` executable is located in the same `JAVA_HOME` as `native-image`, the agent and its debug symbols:
+
+```bash
+$ gdb java
+```
+
+Then set the agent arguments for the java process:
+
+```bash
+(gdb) set args -agentlib:native-image-agent=config-output-dir=./native-image-config -jar app.jar
+```
```

Comment:
FYI you can do this in one line with:

```
gdb --args java -agentlib:native-image-agent=config-output-dir=./native-image-config -jar app.jar
```

Reply:
Yeah I was aware. Do you prefer that style?

Reply:
In my workflow certainly, for the guide I don't really mind, either is fine.

Reply:
Fair enough. If you find it more convenient with your workflow let's go with that. I've updated the PR.